### PR TITLE
Fixes #221 Joss Paper

### LINKED ISSUE #221: Joss Paper
Write a paper for The Journal of Open Source Software to outline the Pi4Micronaut library. 

https://joss.theoj.org/

### PR DESCRIPTION
**Pull request by adelgadoj3**
Created the foundation for the JOSS Submission. We can add or remove sections as needed.

### CODE REVIEW

**Comment by SwindleA** [comment]
First full draft of the paper. More discussion on how the library works needs to be discussed.

**Comment by SwindleA** [comment]
I am still trying to make sure the paper compiles correctly with the references. I just need a review on the contents of the paper.

**Review by ruthvikm — CHANGES_REQUESTED** [review]
Talk about the Raspberry Pi GPIO and how we are providing support for different hardware components.

**Comment by ruthvikm** [comment]
> > Talk about the Raspberry Pi GPIO and how we are providing support for different hardware components. Significance of Pi4Micronaut is missing, explain how Pi4Micronaut application provides API interaction for each component and the different functionalities that we provide for hardware components. Also, talk about python dependency when it comes to Raspberry Pi GPIO interactions.
> 
> @ruthvikm The submission guidelines specifically say that the purpose of the software should be described for a non-specialist audience. So, talking about GPIO, API, and python versus java would be to specific towards programmers. Let me know if you disagree.
> 
> Submission [information:](https://joss.readthedocs.io/en/latest/submitting.html), see the section titled "What should my paper contain?".

@SwindleA - Guidelines says, summary should have a High-Level information focus for non-specialist audience. I think we can use statement of work section to write it up clearly. So that, the reviewers can understand our project. Get into the reviewer shoes and read the paper, you can feel what things are lacking in the paper to show it's significance.

File: joss/paper.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+title: 'Pi4Micronaut'
+tags:
+  - Java
+  - Micronaut
+  - RaspberryPi
+  - Pi4j
+  - GPIO
+authors:
```
Comment (ruthvikm) [inline joss/paper.md:9]: Keep all the members as authors

File: joss/paper.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Raspberry Pi include rat licking behavior @Longley, greenhouse gas effect monitoring system @Shah,
+and many other applications @Jolles. Pi4Micronaut also allows for multiple users to read input from sensors. This
+eliminates the elbowing that may occur when multiple researchers are trying to read input from a sensors. The utilization
+of a Raspberry Pi, eliminates the hassle and cost of needing a full computer for one small sensor. 
+
+
+
+# Availability and Usage
+
+The Pi4Micronaut library package can be found [here](https://central.sonatype.com/artifact/io.github.oss-slu/pi4micronaut-utils).
```
Comment (ruthvikm) [inline joss/paper.md:60]: Make bullet points for the links instead of paragraph.

File: joss/paper.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+The repository can be found [here](https://github.com/oss-slu/Pi4Micronaut). How to use Pi4Micronaut can be found in the documentation [here](https://oss-slu.github.io/Pi4Micronaut/).
+
```
Comment (ruthvikm) [inline joss/paper.md:62]: For usage, talk about our OSS-SLU-CheckIn and Home_Automation projects as example use cases of  our library.

File: joss/paper.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+* Passive Buzzer
+* PIR Motion Sensor
+* Ultrasonic Sensor
+
+
+# Acknowledgments
+
+Thank you to Alex Delgado, Joe Folen, and John Yanev for helping create the first full release of Pi4Micronaut. To the 
+previous team who set the groundwork for us and set us up for the successful release of 1.0. A big thank you to Jeff Brown from the Unity Foundation for overseeing and supporting this project.
+
```
Comment (ruthvikm) [inline joss/paper.md:86]: Talk about Open Source with SLU and who's funding us.

**Review by ruthvikm — CHANGES_REQUESTED** [review]

File: joss/paper.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+# Summary
+
+Pi4Micronaut is an innovative Java library crafted for researchers who aim to create Internet of Things (IoT) 
+applications for their research leveraging the Raspberry Pi platform. This software is designed to run directly on the 
+Raspberry Pi, providing an interface between the high-level Micronaut framework and the low-level hardware control 
+provided by Pi4J. It serves as a vital tool for researchers who wish to use various sensors and electronic components 
+in research. By abstracting the complexity of hardware interactions, Pi4Micronaut allows developers to focus on crafting
+logic and features, making it easier to bring IoT applications from concept to deployment rapidly.
+
+Pi4Micronaut's utilization of the Micronaut framework allows for HTTP requests to interact with the software. These
```
Comment (kate-holdener) [inline joss/paper.md:34]: Maybe change the subject of the sentence here. Something like  "The Micronaut framework, widely used for building web applications and microservices, handles HTTP requests. These requests are then passed to Micronaut's application controller, which is custom code written for each specific web application. The application controller can make calls to the Pi4Micronaut library to interact with the hardware components. Pi4Micronaut library uses Pi4J for low-level hardware interaction. Thus, Pi4Micronaut provides an abstraction layer between Pi4J utilities and Micronaut framework, allowing the developer to focus on the application logic.

Comment (kate-holdener) [reply joss/paper.md:34]: It may be good to create a diagram illustrating this description.

File: joss/paper.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+in research. By abstracting the complexity of hardware interactions, Pi4Micronaut allows developers to focus on crafting
+logic and features, making it easier to bring IoT applications from concept to deployment rapidly.
+
+Pi4Micronaut's utilization of the Micronaut framework allows for HTTP requests to interact with the software. These 
+requests then can be handled application's controller which can use the helper classes provided by Pi4Micronaut. These
+helper classes then interact with the connected hardware components by utilizing Pi4J's utilities. By providing the 
+abstraction between the utilities of Pi4J and the developer, the developer will have less to worry during development.
+Pi4Micronaut being built on Micronaut, allows for the easy creation of a server-like application that handles HTTP requests. 
+These requests can come from anyone on the same Wi-Fi network. This allows for multiple people to interact with the instruments
+that are connected to the Raspberry Pi. For a developer, Pi4Micronaut fills the need of a Java framework for IoT applications.
```
Comment (kate-holdener) [inline joss/paper.md:40]: I would make this a separate paragraph. Again, I suggest to use simpler and more direct wording. With Micronaut and Pi4Micronaut, developers can create web applications to monitor and control their research instruments. Web application is beneficial in cases where multiple researchers need to interact with the same instruments. Web application running on a Raspberry Pi will provide access to the instruments to any user on the same Wi-Fi network. For stricter access control, applications can utilize authentication and authorization mechanisms.

File: joss/paper.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+
+
+
+# Statement of Need
+
+For researchers, a major component of their work is data gathering. While there are many instances when they can simply
```
Comment (kate-holdener) [inline joss/paper.md:47]: Again, I suggest changing the subject of your sentence: Data gathering is a major element of research. There are many research instruments that can be simply connected to a personal computer and utilized as is. However, there are circumstances for when customized instruments need to be developed. 

Then you would proceed to your "The single board Raspberry Pi computer ... " sentence

**Review by kate-holdener — COMMENTED** [review]
I added some suggestions for improvement.

**Comment by kungfuchicken** [comment]
@ruthvikm i don't want to see this go stale. checking in on readiness/progress. any concerns? it looks good to me.

**Review by kungfuchicken — APPROVED** [review]
lgtm;

**Comment by ruthvikm** [comment]
> @ruthvikm i don't want to see this go stale. checking in on readiness/progress. any concerns? it looks good to me.

Working on it. Paper need some minor tweaks. I'll close this pull request by end of this week.

**Comment by ruthvikm** [comment]
Closing this Pull request and deleting 221-joss-paper branch as it used develop branch as base. Working on a new branch to avoid conflicts.

**Pull request closed without merging**